src/runtime/typemanager.cs

```diff
@@ -422,6 +434,198 @@ internal static IntPtr BasicSubType(string name, IntPtr base_, Type impl)
+            internal static int nb_true_divide = 37;
+            internal static int nb_xor = 38;
+            internal static int sq_ass_item = 39;
+            internal static int sq_concat = 40;
+            internal static int sq_contains = 41;
+            internal static int sq_inplace_concat = 42;
+            internal static int sq_inplace_repeat = 43;
+            internal static int sq_item = 44;
+            internal static int sq_length = 45;
+            internal static int sq_repeat = 46;
+            internal static int tp_alloc = 47;
+            internal static int tp_base = 48;
+            internal static int tp_bases = 49;
+            internal static int tp_call = 50;
+            internal static int tp_clear = 51;
+            internal static int tp_dealloc = 52;
+            internal static int tp_del = 53;
+            internal static int tp_descr_get = 54;
+            internal static int tp_descr_set = 55;
+            internal static int tp_doc = 56;
+            internal static int tp_getattr = 57;
+            internal static int tp_getattro = 58;
+            internal static int tp_hash = 59;
+            internal static int tp_init = 60;
+            internal static int tp_is_gc = 61;
+            internal static int tp_iter = 62;
+            internal static int tp_iternext = 63;
+            internal static int tp_methods = 64;
+            internal static int tp_new = 65;
+            internal static int tp_repr = 66;
+            internal static int tp_richcompare = 67;
+            internal static int tp_setattr = 68;
+            internal static int tp_setattro = 69;
+            internal static int tp_str = 70;
+            internal static int tp_traverse = 71;
+            internal static int tp_members = 72;
+            internal static int tp_getset = 73;
+            internal static int tp_free = 74;
+            internal static int nb_matrix_multiply = 75;
+            internal static int nb_inplace_matrix_multiply = 76;
+            internal static int am_await = 77;
+            internal static int am_aiter = 78;
+            internal static int am_anext = 79;
+            internal static int tp_finalize = 80;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct PY_TYPE_SLOT
+        {
+            public long slot; //slot id, from typeslots.h
+            public IntPtr func; //function pointer of the function implementing the slot
+        }
+
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+        internal class PyTypeSpecOffset
+        {
+            static PyTypeSpecOffset()
+            {
+                Type type = typeof(PyTypeSpecOffset);
+                FieldInfo[] fi = type.GetFields();
+                int size = IntPtr.Size;
+                for (int i = 0; i < fi.Length; i++)
+                {
+                    fi[i].SetValue(null, i * size + TypeOffset.ob_size);
+                }
+            }
```

Comment:
As long as this structure does not differ between Python versions (which I am hoping it is not designed to), you can just set these fields to be constant or `static readonly` values, and not use reflection at all.

Reply:
@lostmsu I think IntPtr.Size depends on the platform.  I'm getting crashes in PyType_FromSpec and I suspect the integer fields are only 4 bytes so using IntPtr.Size to increment each field in this struct would be wrong.  If my suspicion turns out to be correct then I wonder how things are working in the other struct layout objects which assume each field is a pointer.  

Either way my code should change..  I'll keep you posted

Reply:
@koubaa you can use `IntPtr.Size` in `static readonly` fields.

BTW, does type spec allow us to drop `TypeOffset` class?

Reply:
@lostmsu Good point.  If the arithmetic isn't too obtuse I might go that way

Yes, we can drop TypeOffset once all usages of it has migrated to PyType_FromSpec, which is my intention before landing this change